include/oneapi/mkl/sparse_blas/detail/onemkl_sparse_blas_backends.hxx

```diff
@@ -20,72 +20,198 @@
+                                   sycl::buffer<indexType, 1> col_ind,
+                                   sycl::buffer<dataType, 1> val);
+template <typename dataType, typename indexType>
+ONEMKL_EXPORT void init_coo_matrix(sycl::queue &queue, matrix_handle_t *p_smhandle,
+                                   std::int64_t num_rows, std::int64_t num_cols, std::int64_t nnz,
+                                   index_base index, indexType *row_ind, indexType *col_ind,
+                                   dataType *val);
+
+template <typename dataType, typename indexType>
+ONEMKL_EXPORT void set_coo_matrix_data(sycl::queue &queue, matrix_handle_t smhandle,
+                                       std::int64_t num_rows, std::int64_t num_cols,
+                                       std::int64_t nnz, index_base index,
+                                       sycl::buffer<indexType, 1> row_ind,
+                                       sycl::buffer<indexType, 1> col_ind,
+                                       sycl::buffer<dataType, 1> val);
+template <typename dataType, typename indexType>
+ONEMKL_EXPORT void set_coo_matrix_data(sycl::queue &queue, matrix_handle_t smhandle,
+                                       std::int64_t num_rows, std::int64_t num_cols,
+                                       std::int64_t nnz, index_base index, indexType *row_ind,
+                                       indexType *col_ind, dataType *val);
+
+// CSR matrix
+template <typename dataType, typename indexType>
+ONEMKL_EXPORT void init_csr_matrix(sycl::queue &queue, matrix_handle_t *p_smhandle,
+                                   std::int64_t num_rows, std::int64_t num_cols, std::int64_t nnz,
+                                   index_base index, sycl::buffer<indexType, 1> row_ptr,
+                                   sycl::buffer<indexType, 1> col_ind,
+                                   sycl::buffer<dataType, 1> val);
+template <typename dataType, typename indexType>
+ONEMKL_EXPORT void init_csr_matrix(sycl::queue &queue, matrix_handle_t *p_smhandle,
+                                   std::int64_t num_rows, std::int64_t num_cols, std::int64_t nnz,
+                                   index_base index, indexType *row_ptr, indexType *col_ind,
+                                   dataType *val);
+
+template <typename dataType, typename indexType>
+ONEMKL_EXPORT void set_csr_matrix_data(sycl::queue &queue, matrix_handle_t smhandle,
+                                       std::int64_t num_rows, std::int64_t num_cols,
+                                       std::int64_t nnz, index_base index,
+                                       sycl::buffer<indexType, 1> row_ptr,
+                                       sycl::buffer<indexType, 1> col_ind,
+                                       sycl::buffer<dataType, 1> val);
+template <typename dataType, typename indexType>
+ONEMKL_EXPORT void set_csr_matrix_data(sycl::queue &queue, matrix_handle_t smhandle,
```

Comment:
In the previous version, we had:
```cpp
template <typename fpType, typename intType>
ONEMKL_EXPORT std::enable_if_t<detail::are_fp_int_supported_v<fpType, intType>> set_csr_data(
    sycl::queue &queue, matrix_handle_t handle, intType num_rows, intType num_cols, intType nnz,
    index_base index, sycl::buffer<intType, 1> &row_ptr, sycl::buffer<intType, 1> &col_ind,
    sycl::buffer<fpType, 1> &val);
```
Don't we need the `std::enable_if_t<detail::are_fp_int_supported_v<fpType, intType>>` to still be there in the new version?

Reply:
In `sparse_blas/detail/sparse_blas_ct.hxx`, for instance, I see you still have the `enable_if_t` going in.